### DISCUSSION TITLE
502: BAD_GATEWAY  NO_RESPONSE_FROM_FUNCTION

### DISCUSSION THREAD

**Discussion by rafael-legyo** [vercel/next.js#20073] [Help]
This error presented to 4/5 people who tested it.
Then they refreshed and it worked correctly, and later it didn't show any more errors, in several tests.

That is, it apparently happens the first time you access the route / checkout

![Screen Shot 2020-12-09 at 20 20 37](https://user-images.githubusercontent.com/48858913/101818719-08973500-3b03-11eb-8d35-a81b0ba3f672.png)

![Screen Shot 2020-12-09 at 20 20 31](https://user-images.githubusercontent.com/48858913/101818714-059c4480-3b03-11eb-8c08-63c1b8cf1763.png)

example url checkout: https://prepta-2-0-frontend.vercel.app/checkout?id=3c07e913-21c1-40da-8856-6203fc8a8c9b

`export async function getServerSideProps(context) {
  const productID = context.query.id;
  const res = await fetch(
    `https://prepta-2-0.herokuapp.com/produtos/${productID}`
  );
  const product = await res.json();

  return {
    props: {
      product,
    },
  };
}`

What could it be?

**Comment by waptik** [comment]
It seems you are on the hobby plan. So according to https://vercel.com/docs/platform/limits#general-limits, when you look at **Serverless Function Duration (Seconds)**, you'll find out that the value is 10. Which means that if your api route takes more than 10 secends to respond, you'll get an error. If you don't want that to happen again, upgrade to a higher plan
Since you are using heroku, it'll do you good if you setup a cache mechanism.

**Comment by rafael-legyo** [comment]
@waptik

makes sense!

I have no idea how I would do this on heroku! I'm a beginner.

But would using SWR already help?

**Reply by waptik** [reply]
I'm not sure. From my point of view, i'd say you have to work on heroku side as that's where the slow request is coming from. SWR won't have much impact if the server response is slow.
